frame/system/src/lib.rs

```diff
@@ -225,6 +226,21 @@ pub trait Trait: 'static + Eq + Clone {
 
 	/// Migrate an account.
 	type MigrateAccount: MigrateAccount<Self::AccountId>;
+
+	/// The central instance that dispatches all calls.
+	///
+	/// The `RootDispatcher` is the central dispatcher and all other dispatchers should
+	/// eventually delegate to it. The default runtime uses the `RootDispatcher` for all
+	/// modules that do dispatching themselves.
+	///
+	/// However, a runtime implementer can choose to provide a custom dispatcher to some
+	/// of the modules that is different from the `MainDispatcher´. When doing this it is
+	/// important to delegate to the `RootDispatcher`. There might be use cases where the
+	/// delegation is unwanted and the custom dispatcher dispatches the `Dispatchable`
+	/// itself. It is important to understand that by doing this the dispatch becomes
+	/// invisible to the rest of the runtime machinery which relies on the `RootDispatcher`
```

Comment:
Anytime there is something that is "important to understand" written in docs then I shivers down my spine. 

This "API" (and I would use the term generously here, as it's far from codified) is dangerous.

Reply:
I agree with you. I assumed that we need a way to enforce the correct usage which triggered this [discussion](https://github.com/paritytech/substrate/pull/5130#discussion_r392571834). Bottom line was that the soft API or "escape hatch" was intentionally. You are rightfully unhappy about it.